lib/pbench/server/auth/__init__.py

```diff
@@ -123,24 +126,30 @@ def get(
         Returns:
             Response from the request.
         """
-        return self._method("GET", path, None, headers=headers, **kwargs)
+        return self._method("GET", path, None, None, headers=headers, **kwargs)
 
     def post(
-        self, path: str, data: Dict, headers: Optional[Dict] = None, **kwargs
+        self,
+        path: str,
+        data: Union[Dict, str] = None,
+        json: Optional[Dict] = None,
+        headers: Optional[Dict] = None,
+        **kwargs,
     ) -> requests.Response:
```

Comment:
The type hint for `data` should indicate that it is `Optional`, given that you've set the default to something which is neither of the two types.  😉 

In the original code, `data` was only a `Dict` and not a `str`; should you be adding `str` here, now that we have `json`?

Given that `data`, `json`, and `headers` are passed straight through to `_method()`, consider omitting them altogether from the function signature here:  if they are specified by the caller (as keyword arguments), then they'll be placed in `kwargs` which you're already providing to `_method()`.

Reply:
I think `data: Any` would be most correct.

Reply:
> In the original code, `data` was only a `Dict` and not a `str`; should you be adding `str` here, now that we have `json`

It can be a string I think I am sending the data in the string format currently with `json.dumps`. I think it can take any format isn't it?

Reply:
> It can be a string I think I am sending the data in the string format currently with `json.dumps`.

My impression is that, if the caller is tempted to stringify their JSON, the shouldn't be using the `data` parameter but should be using the `json` parameter instead.  And, therefore, we should _not_ be **adding** `str` to the permitted types for `data`.  (I thought you got rid of your `json.dumps()` call.)

> I think it can take any format isn't it?

This is our code that we're talking about here (not the underlying Requests package), and so we can put any limitations on it which we think are appropriate.  Given the "interesting" behaviors of `data` and the fact that we almost always use JSON bodies, I think it would be a Good Thing™ to try to prevent stringified JSON from being passed via `data`.